### PR TITLE
Add ci-operator config and job for cri-o

### PR DESCRIPTION
**Pull request by umohnani8**
Add yaml files to config and jobs for cri-o release-1.13
branch so that we can trigger the e2e-aws tests on cri-o
PRs.

Signed-off-by: Urvashi Mohnani <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@mrunalp @smarterclayton PTAL

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        - --sentry-dsn-path=/etc/sentry-dsn/ci-operator
+        - --target=[images]
+        command:
+        - ci-operator
+        env:
+        - name: CONFIG_SPEC
+          valueFrom:
+            configMapKeyRef:
+              key: cri-o-cri-o-release-1.13.yaml
+              name: ci-operator-4.1-configs
```
Comment (umohnani8) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: @smarterclayton I am not sure if this should be the value of the name or if it should be something like `ci-operator-1.13-configs`.

Comment (umohnani8) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: nvm, looks like the generator fixed it

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+postsubmits:
```
Comment (smarterclayton) [inline ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: You shouldn't need this, remove the promotion config from your job (remove `promotion:`) and then regen and it should go away.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- as: unit
+  commands: make testunit
+  container:
+    from: src
+- as: e2e-aws
+  commands: TEST_SUITE=openshift/conformance/parallel run-tests
+  openshift_installer:
+    cluster_profile: aws
+    upgrade: false
+- as: launch-aws
```
Comment (smarterclayton) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml]: you'll need to make this an optional job (after generating) by adding the correct "optional" attributes in presubmits.yaml.  See what origin launch-aws does. (required: false, optional: true, I think)

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml (hunk 2 above)
Comment (umohnani8) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: I removed `promotion:` from the config yaml and regenerated, but the postsubmits did not go away.

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml (hunk 3 above)
Comment (umohnani8) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml]: done

File: ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml (hunk 2 above)
Comment (smarterclayton) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: delete the new files under ci-operator/jobs/* that got created, then run regenerate

Comment (umohnani8) [reply ci-operator/jobs/cri-o/cri-o/cri-o-cri-o-release-1.13-postsubmits.yaml]: got it, thanks! done

**Comment by umohnani8** [comment]
Waiting https://github.com/cri-o/cri-o/pull/2449 to get in for the rehearse tests here to pass

File: ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+base_images:
+  base:
+    name: "4.1"
```
Comment (stevekuznetsov) [inline ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml]: 4.2?

Comment (umohnani8) [reply ci-operator/config/cri-o/cri-o/cri-o-cri-o-release-1.13.yaml]: This is for cri-o 1.13, so that should be for 4.1. cri-o 1.14 will be for 4.2.

**Comment by umohnani8** [comment]
@smarterclayton the e2e-aws test suite is failing on 2 prometheus tests https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_release/3962/rehearse-3962-pull-ci-cri-o-cri-o-release-1.13-e2e-aws/7.
From one of the tests at https://github.com/openshift/origin/blob/master/test/extended/prometheus/prometheus_builds.go#L120, I see that is not getting the "success" results it needs to pass.
What exactly are the prometheus tests looking for? I am not sure about what it is not getting from cri-o. Metrics are enabled in cri-o and the port is set to 9537.

**Comment by petr-muller** [comment]
@stevekunetsov ouch, the `release-1.13` branch is conflicting with the openshift branching scheme...

**Comment by smarterclayton** [comment]
@adambkaplan do you know why these would fail?

**Comment by adambkaplan** [comment]
@smarterclayton no idea - that method is shared between the normal prometheus tests and the build metrics test.

@brancz can you shed some light here?

**Comment by brancz** [comment]
The failure is that there is an alert firing, that shouldn't be firing, and missing the crio job.

The discovery for crio is configured here: https://github.com/openshift/cluster-monitoring-operator/blob/5092b55844dfec03d61f0c767f01da887e95f40c/assets/prometheus-k8s/service-monitor-kubelet.yaml#L32-L48

I'm downloading the Prometheus dump to understand more.

**Comment by brancz** [comment]
Yep looks like it's not a discovery problem, but it really can't be scraped for some reason, so the failed CI is legit:

Screenshot from 2019-06-12 14-17-45

**Comment by umohnani8** [comment]
Tests are green! The sharded-config test error is saying
```
You are adding a file that is not covered by the automatic upload.
[ERROR] Contact an administrator to resolve this issue.
```
@smarterclayton what do I need to do to make this pass?

**Comment by petr-muller** [comment]
> Tests are green! The sharded-config test error is saying
> ...
> what do I need to do to make this pass?

@umohnani8 Some of your config files are not covered by the config-updater config [here](https://github.com/openshift/release/blob/04e44de48dd93377cab061c502b96ffd248e7162/cluster/ci/config/prow/plugins.yaml#L885), meaning they wouldn't be automatically uploaded to the cluster. You need to add regexes covering your configs there, for *both* `ci-operator/config/...` and `ci-operator/jobs/...` paths.

`...-misc` configmaps look like a proper destination for these, like [job CM here](https://github.com/openshift/release/blob/04e44de48dd93377cab061c502b96ffd248e7162/cluster/ci/config/prow/plugins.yaml#L905) and [ci-op CM here](https://github.com/openshift/release/blob/04e44de48dd93377cab061c502b96ffd248e7162/cluster/ci/config/prow/plugins.yaml#L941)

*Note:* The config-updater config needs to be merged in a separate PR first

**Comment by umohnani8** [comment]
@petr-muller got it, thanks! Opened https://github.com/openshift/release/pull/4082 for the config-updater.

**Comment by umohnani8** [comment]
@petr-muller all tests are green now :)

**Pull request merged**